### ISSUE TITLE
Incorrect internal port configuration in Minecraft data

### ISSUE THREAD

**Issue by fisher60** [itzg/docker-minecraft-server#1958]
### Describe the problem

In docker compose, when mapping the external/internal ports for the container, the config is breaking and causing a `connection refused` error on the Minecraft client. This appears to be due to the configuration files, such as server.properties, using the *external* port mapping, rather than the internal port mapping.

I.e in the example compose file pasted below, I am attempting to bind my external ports (25567 and 25577 respectively), that are defined in my `.env` file, to the internal, default minecraft ports. This causes the server to expect the *internal* port of `25567`, therefore not allowing connections to the server since the container is expecting to communicate internally over 25575.

The temporary solution is to always ensure the external and internal ports are identical, however this should not be necessary.

### Container definition

```yaml
services:
  mc:
    image: itzg/minecraft-server
    ports:
      - ${SERVER_PORT}:25565
      - ${RCON_PORT}:25575
    env_file:
      - .env
    environment:
      EULA: "TRUE"
      TYPE: "PAPER"
    tty: true
    stdin_open: true
    restart: unless-stopped
    volumes:
      - ./data:/data
```

### Container logs

Logs appear healthy, no error messages received.

**Comment by fisher60** [comment]
The `server.properties` file is the default that is generated from running `docker compose up --build`. 

I realized just now that it is possible the image is using my environment variable `SERVER_PORT` to do some type of internal configuration? I have been using this image for some time and recently went to create a new server using an older `.env` file, was there an update that happens to now utilize the `SERVER_PORT` env var I specified in my old configuration?
